addons/product/models/res_partner.py

```diff
@@ -20,7 +20,7 @@ def _compute_product_pricelist(self):
         company = self.env.context.get('force_company', False)
         res = self.env['product.pricelist']._get_partner_pricelist_multi(self.ids, company_id=company)
         for p in self:
-            p.property_product_pricelist = res.get(p.id)
+            p.property_product_pricelist = res.get(p._origin.id)
```

Comment:
Maybe
```
product.property_product_pricelist = res.get(product.id if product.ids else product._origin.id)
```

Reply:
This solution is not good, because on p.ids we have the list of the ids that build the NewId object. So it will be evaluated as True and that will give us the same problem.

Reply:
Normally on cache records, ids is a void list [] but id is a newId (which is why you have an issue using directly p.id).

You may also use if not isinstance(p.id, models.NewId) . But I would prefer the first solution.